src/liballoc/boxed.rs

```diff
@@ -127,24 +127,38 @@ impl<T: ?Sized> Box<T> {
     ///
     /// After calling this function, the raw pointer is owned by the
     /// resulting `Box`. Specifically, the `Box` destructor will call
-    /// the destructor of `T` and free the allocated memory. Since the
-    /// way `Box` allocates and releases memory is unspecified, the
-    /// only valid pointer to pass to this function is the one taken
-    /// from another `Box` via the [`Box::into_raw`] function.
+    /// the destructor of `T` and free the allocated memory. For this
+    /// to be safe, the memory must have been allocated in the precise
+    /// way that `Box` expects, namely, using the global allocator
+    /// with the correct [`Layout`] for holding a value of type `T`. In
+    /// particular, this will be satisfied for a pointer obtained
+    /// from a previously existing `Box` using [`Box::into_raw`].
+    ///
+    /// # Safety
     ///
     /// This function is unsafe because improper use may lead to
     /// memory problems. For example, a double-free may occur if the
     /// function is called twice on the same raw pointer.
     ///
-    /// [`Box::into_raw`]: struct.Box.html#method.into_raw
-    ///
     /// # Examples
-    ///
+    /// Recreate a `Box` which was previously converted to a raw pointer using [`Box::into_raw`]:
     /// ```
     /// let x = Box::new(5);
     /// let ptr = Box::into_raw(x);
     /// let x = unsafe { Box::from_raw(ptr) };
     /// ```
+    /// Manually create a `Box` from scratch by using the global allocator:
+    /// ```
+    /// use std::alloc::{Layout, alloc};
+    ///
+    /// let ptr = unsafe{ alloc(Layout::new::<i32>()) } as *mut i32;
+    /// unsafe{ *ptr = 5; }
+    /// let x = unsafe{ Box::from_raw(ptr) };
```

Comment:
Instead of an `unsafe` on each line, could this just be wrapped in one block? Additionally could "rustfmt be run over the code examples" here? I think that there's typically a space after the `unsafe` word, for example.

Reply:
> rustfmt be run over the code examples

[This can actually be done with an unstable rustfmt option](https://github.com/rust-lang/rustfmt/issues/3348)

Reply:
Done! And thanks for the tip on the new rustfmt feature; it worked!